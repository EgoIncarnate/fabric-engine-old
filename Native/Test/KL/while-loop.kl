/*
 *  Copyright 2010-2012 Fabric Technologies Inc. All rights reserved.
 */

operator entry()
{
  Size i=0;
  while ( i < 10 )
    report ++i;
}
