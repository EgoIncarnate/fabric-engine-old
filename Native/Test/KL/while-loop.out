GlobalList
  Function( 'entry' ('entry'), void )
    ParamList
    CompoundStatement
      StatementList
        StatementList
          AssignedVarDecl( VarDecl( 'i', Size ), ConstSize( 0 ) )
            ConstSize( 0 )
        CStyleLoop
          BinOp( LT )
            Var('i')
            ConstSize( 10 )
          Report
            UniOp( prefix ++ )
              Var('i')
1
2
3
4
5
6
7
8
9
10
